talk/morelanguage/ranges.tex

```diff
@@ -0,0 +1,93 @@
+\subsection[range]{Ranges}
+
+\begin{frame}[fragile]
+  \frametitlecpp[20]{Ranges}
+  \begin{block}{The range concept}
+    \begin{itemize}
+    \item A range is a group of items that you can iterator over
+    \item It provides \mintinline{cpp}{begin} and \mintinline{cpp}{end} functions
```

Comment:
Hmm, it does not need to provide `begin` and `end` in the sense of member functions. A range `r` needs to have `std::ranges::begin(r)` "work" and vice-versa for `end`. I wonder whether we should rephrase the sentence to something like:

```suggestion
    \item It must provide iterators to its beginning and end, obtainable via \mintinline{cpp}{std::range::begin} and \mintinline{cpp}{std::range::end}
```